src/__experimental__/components/form/form.component.js

```diff
@@ -502,7 +518,22 @@ FormWithoutValidations.propTypes = {
   warningCount: PropTypes.number,
 
   /** The total number of infos present in the form */
-  infoCount: PropTypes.number
+  infoCount: PropTypes.number,
+
+  /** Strores the state of controlled inputs */
+  formInputs: PropTypes.shape({
+    value: PropTypes.string,
+    name: PropTypes.string
+  }),
+
+  /** The action for the default form submission of controlled inputs */
+  formAction(props, propName) {
+    if ((!props.onSubmit && (props[propName] === undefined || typeof (props[propName]) !== 'string'))) {
```

Comment:
could you extract these conditionals to increase readability?

Reply:
Agreed 👍

Reply:
Addressed in https://github.com/Sage/carbon/pull/2175/commits/27e1fca2dea6feee76c60bbeda6e239da71a4f5a